src/Compilers/CSharp/Test/Semantic/SourceGeneration/GeneratorDriverTests.cs

```diff
@@ -890,6 +891,42 @@ class C { }
             AssertEx.Equal(results.Diagnostics, fullDiagnostics);
         }
 
+        [Fact]
+        public void Cancellation_During_Initialization_Doesnt_Report_As_Generator_Error()
+        {
+            var source = @"
+class C 
+{
+}
+";
+            var parseOptions = TestOptions.Regular;
+            Compilation compilation = CreateCompilation(source, options: TestOptions.DebugDll, parseOptions: parseOptions);
+            compilation.VerifyDiagnostics();
+
+            Assert.Single(compilation.SyntaxTrees);
+
+            CancellationTokenSource cts = new CancellationTokenSource();
+
+            var testGenerator = new CallbackGenerator(
+                onInit: (i) => { cts.Cancel(); },
+                onExecute: (e) => { }
+                );
+
+            // test generator cancels the token. Check that the call to this generator doesn't make it look like it errored.
+            var testGenerator2 = new CallbackGenerator2(
+                onInit: (i) => { i.CancellationToken.ThrowIfCancellationRequested(); },
```

Comment:
Assert that i.CancellatonToken has been cancelled? Otherwise this is implicitly assuming that generators are initialized in order (which is the case today but not guaranteed by API). But if it were go change order then the test may be invalidated without a failure.